hello_world/views.py

```diff
@@ -1,3 +1,25 @@
-from django.shortcuts import render
+from hello_world.models import SearchHistory, User
+from hello_world.serializers import SearchHistorySerializer, UserSerializer
+from rest_framework import generics
 
-# Create your views here.
+
+class SearchHistoryList(generics.ListCreateAPIView):
+    queryset = SearchHistory.objects.all()
+    serializer_class = SearchHistorySerializer
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+
+class SearchHistoryDetail(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
RetrieveUpdateDestroyAPIView - probably not a good view to use
Try RetrieveAPIView